pkg/server/delta/v3/server.go

```diff
@@ -160,14 +160,17 @@ func (s *server) processDelta(str stream.DeltaStream, reqCh <-chan *discovery.De
 			if !ok {
 				// Initialize the state of the stream.
 				// Since there was no previous state, we know we're handling the first request of this type
-				// so we set the initial resource versions if we have any, and also signal if this stream is in wildcard mode.
+				// so we set the initial resource versions if we have any.
+				// We also set the stream as wildcard based on its legacy meaning (no resource subscribed).
```

Comment:
This is a bit misleading IMO. The legacy meaning of wildcard was ALL resources subcribed but we never kept them in the resources subscription list.

Reply:
I will rephrase. By no resource subscribed I meant none sent in `req.GetResourceNamesSubscribe()`